packages/flutter_tools/lib/src/ios/simulators.dart

```diff
@@ -403,57 +405,81 @@ class IOSSimulator extends Device {
     }
   }
 
-  @override
-  bool isSupported() {
-    if (!Platform.isMacOS) {
-      _supportMessage = "Not supported on a non Mac host";
-      return false;
-    }
-
-    // Step 1: Check if the device is part of a blacklisted category.
-    //         We do not support WatchOS or tvOS devices.
-
-    RegExp blacklist = new RegExp(r'Apple (TV|Watch)', caseSensitive: false);
+  static bool _isWatchOrTVSimulator(String name) {
+    return (new RegExp(r'Apple (TV|Watch)', caseSensitive: false)).hasMatch(name);
+  }
 
-    if (blacklist.hasMatch(name)) {
-      _supportMessage = "Flutter does not support either the Apple TV or Watch. Choose an iPhone 5s or above.";
-      return false;
-    }
+  static bool _isiPhoneSimulator(String name) {
+    return (new RegExp(r'iPhone', caseSensitive: false)).hasMatch(name);
+  }
 
-    // Step 2: Check if the device must be rejected because of its version.
-    //         There is an artitifical check on older simulators where arm64
-    //         targetted applications cannot be run (even though the
-    //         Flutter runner on the simulator is completely different).
+  static bool _isiPadSumulator(String name) {
+    return (new RegExp(r'iPad', caseSensitive: false)).hasMatch(name);
+  }
 
-    RegExp versionExp = new RegExp(r'iPhone ([0-9])+');
+  static bool _isSupportediPhoneSimulator(String name) {
+    RegExp versionExp = new RegExp(r'iPhone ([0-9])+', caseSensitive: false);
```

Comment:
It looks like in `IOSSimulatorUtils.getAttachedDevices()`, you have access to the `SimDevice` that's used to instantiate the `IOSSimulator`.  The `SimDevice` has a `category` field that looks like the right thing to check for compatibility (instead of the device name).